virttest/utils_misc.py

```diff
@@ -2731,22 +2731,68 @@ def set_winutils_letter(session, cmd, label="WIN_UTILS"):
     return cmd
 
 
-def format_windows_disk(session, did, mountpoint=None, size=None, fstype="ntfs"):
+def list_linux_guest_disks(session, partition=False):
+    """
+    List all disks OR disks with no partition in linux guest.
+
+    :param session: session object to guest
+    :param partition: if true, list all disks; otherwise,
+                      list only disks with no partition.
+    :return: the disks set.
+    """
+    cmd = "ls /dev/[vhs]d*"
+    if not partition:
+        cmd = "%s | grep -v [0-9]$" % cmd
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        raise exceptions.TestFail()
```

Comment:
Better add some info why it fails. Maybe
raise exceptions.TestFail("Get device failed with output %s" % output)